tests/test_downloader_handlers.py

```diff
@@ -437,6 +437,8 @@ class S3AnonTestCase(unittest.TestCase):
         import boto
     except ImportError:
         skip = 'missing boto library'
+    if six.PY3:
+        skip = 'S3 not supported on Py3'
```

Comment:
I think that porting s3 was originally postponed because of boto dependency, which does not support python 3. There is boto3 which is now stable and has py3 support, so s3 support should be possible.